addons/mrp/models/mrp_workorder.py

```diff
@@ -324,13 +315,28 @@ def _compute_color(self):
         for order in (self - late_orders):
             order.color = 2
 
-    @api.onchange('date_planned_start', 'duration_expected')
+    @api.onchange('date_planned_start', 'duration_expected', 'workcenter_id')
     def _onchange_date_planned_start(self):
-        if self.date_planned_start and self.duration_expected:
-            self.date_planned_finished = self.workcenter_id.resource_calendar_id.plan_hours(
-                self.duration_expected / 60.0, self.date_planned_start,
-                compute_leaves=True, domain=[('time_type', 'in', ['leave', 'other'])]
-            )
+        if self.date_planned_start and self.duration_expected and self.workcenter_id:
+            self.date_planned_finished = self._calculate_date_planned_finished()
+
+    def _calculate_date_planned_finished(self, date_planned_start=False):
+        return self.workcenter_id.resource_calendar_id.plan_hours(
+            self.duration_expected / 60.0, date_planned_start or self.date_planned_start,
+            compute_leaves=True, domain=[('time_type', 'in', ['leave', 'other'])]
+        )
+
+    @api.onchange('date_planned_finished')
+    def _onchange_date_planned_finished(self):
+        if self.date_planned_start and self.date_planned_finished:
+            self.duration_expected = self._calculate_duration_expected()
+
+    def _calculate_duration_expected(self, date_planned_start=False, date_planned_finished=False):
+        interval = self.workcenter_id.resource_calendar_id.get_work_duration_data(
```

Comment:
What if the record has no start and end date, and we only write one? we get an error... 
if dt.tzinfo:
AttributeError: 'bool' object has no attribute 'tzinfo'

Reply:
@MerlinGuillaume

Reply:
i created a new commit to v13 this time, so you dont need to create a new one 
https://github.com/odoo/odoo/pull/101715

Reply:
Could you please provide the steps to reproduce this error ?

Reply:
yh,
1- Open an MO with workorders
2- Unplan
3- Add a Plan start date to one of the workorders (the onchage will add the end date)
4- Remove the end date (that odoo added with the onchange)
5- Save

Reply:
Hi, unless I'm mistaken, the error only happens in v14 so there's no need to create a PR in v13. Moreover I think a test would be great in this case so I think it's easier if I take over. Could you please create a ticket to track this new issue more clearly ? You can then send it to me

Reply:
looks like u are avoiding my PR but ok :)
About the ticket i just gave up on your support team since the reponse is always the same "It is expected behavior" or "We take in consideration to the next version"